src/hotspot/cpu/x86/x86.ad

```diff
@@ -9184,6 +9394,7 @@ instruct mask_opers_evex(kReg dst, kReg src1, kReg src2, kReg kscratch) %{
     const MachNode* mask2 = static_cast<const MachNode*>(this->in(this->operand_index($src2)));
     assert(0 == Type::cmp(mask1->bottom_type(), mask2->bottom_type()), "");
     uint masklen = Matcher::vector_length(this);
+    masklen = masklen < 16 && !VM_Version::supports_avx512dq() ? 16 : masklen;
```

Comment:
Good to add parenthesis here: 
(masklen < 16 && !VM_Version::supports_avx512dq())